### PR TITLE
Implement moment function

### PR DESCRIPTION
**Pull request by ibusko**
As per #745 

This is for now being implemented as a function, not as a method in the `Spectrum1D` class. The rationale is that, in `specutils`, measurement and processing operations are implemented as functions in top-level packages such as `analysis`, `fitting`, and `manipulation`, and not as methods in `Spectrum1D` or `OneDSpectrumMixin`. I am not sure we should break this pattern now.

The `moment` method in `spectral_cube` accepts a parameter (how: cube | slice | ray | auto) that is meant to select specialized cube addressing patterns that optimize the calculation for certain cube and memory configurations. We leave this out of this PR for now (it's using the "cube" mode).

### CODE REVIEW

**Review by rosteen — APPROVED** [review]
This looks good to me. The tests are passing for me locally, but it seems like there have been some connection errors in the tests across a few PRs in the Github actions tests. I still need to familiarize myself with the new machinery, not sure if it's just a  matter of kicking them off again or fixing something in the test infrastructure.

**Comment by ibusko** [comment]
I’ve seen all these “canceled” test runs in several PRs, not just mine. It’s like something is timing out.

Sent from my iPhone

On Jan 15, 2021, at 11:48, rosteen <notifications@github.com> wrote:

﻿

@rosteen approved this pull request.

This looks good to me. The tests are passing for me locally, but it seems like there have been some connection errors in the tests across a few PRs in the Github actions tests. I still need to familiarize myself with the new machinery, not sure if it's just a matter of kicking them off again or fixing something.

**Comment by nmearl** [comment]
Are there plans to include some documentation either with this, or in a follow up PR? If not, I'd also suggest opening an issue so we can be sure to address it eventually (preferably before the next release).

**Comment by rosteen** [comment]
I'll open an issue to create a "so you want to do SpectralCube things" section to the docs addressing this and the other related PRs.

**Review by PatrickOgle — APPROVED** [review]
Tested moment order=0 and order=1 on manga cube and got sensible results.  Error regarding masks is fixed.  Ready to merge.

File: specutils/analysis/moment.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+
+def moment(spectrum, regions=None, order=0, axis=-1):
+    """
+    Estimate the moment of the spectrum.
+
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
```
Comment (eteq) [inline specutils/analysis/moment.py:21]: Will this also work on `SpectrumCollection`? *in principle* it should, and maybe we should add a test if it does (and if not make an issue to fix it later?)

**Review by eteq — CHANGES_REQUESTED** [review]
One quick comment that should be an easy either "make a test" or "make an issue" solution, otherwise LGTM!

File: specutils/analysis/moment.py (hunk 1 above)
Comment (ibusko) [reply specutils/analysis/moment.py:21]: I verified now, it does not work with `SpectrumCollection`. I created issue #772 .

Comment (eteq) [reply specutils/analysis/moment.py:21]: Ok, looks good - we can deal with it later then.  Will merge!

**Review by eteq — APPROVED** [review]

**Pull request merged**